system/CMSIS/include/LPC17xx.h

```diff
@@ -1021,6 +1022,7 @@ typedef struct
 #define LPC_GPIO2_BASE        (LPC_GPIO_BASE + 0x00040)
 #define LPC_GPIO3_BASE        (LPC_GPIO_BASE + 0x00060)
 #define LPC_GPIO4_BASE        (LPC_GPIO_BASE + 0x00080)
+#define LPC_GPIO5_BASE        (LPC_GPIO_BASE + 0x000A0)
```

Comment:
GitHub won't let me comment on random lines, so I have to put it here.

LPC_GPIO_BASE is 0x2009C000 for 176x chips, but appears to be at 0x20098000 for 178x.

This base address is defined in several different files throughout the repo. We will need to figure out how to handle that dynamically rather than assuming it is a hard-coded value.

I don't know if other base addresses are impacted as well.

The LPC 177x software has headers for the two families in it, which define their base addresses, we would need something like that, to allow the same code to work with both families of chips.
Here is an unofficial dump of it somewhere on github. Look at the chip_lpc* files.
https://github.com/ciaa/firmware_v2/tree/master/modules/lpc1769/chip/inc